compat/cpu.h

```diff
@@ -9,23 +9,36 @@
 #ifndef SQUID_COMPAT_CPU_H
 #define SQUID_COMPAT_CPU_H
 
-#if HAVE_ERRNO_H
-#include <errno.h> /* for ENOTSUP */
-#endif
 #if HAVE_SCHED_H
 #include <sched.h>
 #endif
 
-#if !HAVE_CPU_AFFINITY
-/* failing replacements to minimize the number of if-HAVE_CPU_AFFINITYs */
-#if !HAVE_CPU_SET_T
+#if !HAVE_SCHED_SETAFFINITY
+
+#if HAVE_STDBOOL_H
+#include <stdbool.h>
+#endif
+#if HAVE_SYS_PARAM_H
+#include <sys/param.h>
+#endif
+#if HAVE_SYS_CPUSET_H
+#include <sys/cpuset.h>
+#if HAVE_CPUSET_T
+typedef cpuset_t cpu_set_t;
+#endif
+#define HAVE_CPU_AFFINITY 1
+inline void
+CpuAnd(cpu_set_t *destset, const cpu_set_t *srcset1, const cpu_set_t *srcset2)
+{
+    cpuset_t desttmp;
+    CPU_COPY(srcset1, &desttmp);
+    CPU_AND(&desttmp, srcset2);
+    CPU_COPY(&desttmp, destset);
```

Comment:
Do we have to introduce a temporary CPU set here?

```suggestion
    CPU_COPY(srcset1, destset);
    CPU_AND(destset, srcset2);
}
```